flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/calcite/FlinkPlannerImpl.scala

```diff
@@ -163,7 +163,35 @@ class FlinkPlannerImpl(
         sqlValidator.getCatalogReader.unwrap(classOf[CalciteCatalogReader]),
         cluster,
         convertletTable,
-        sqlToRelConverterConfig)
+        sqlToRelConverterConfig) {
+        // override convertFrom() to support flexible Temporal Table Syntax,
+        // this can be revert once FLINK-16579(Upgrade Calcite version to 1.23) resolved.
+        val relBuilder = config.getRelBuilderFactory.create(cluster, null)
+
+        override def convertFrom(bb: SqlToRelConverter#Blackboard, from: SqlNode): Unit = {
```

Comment:
Create a JIRA issue to remove this overriding once we bump up Calcite version. And add comment above this method with the JIRA id.